store/store.go

```diff
@@ -420,11 +421,8 @@ type sectionResults struct {
 	} `json:"_embedded"`
 }
 
-// The fields we are interested in
-var detailFields = getStructFields(snapDetails{})
-
-// The fields we are interested in for snap.ChannelSnapInfos
-var channelSnapInfoFields = getStructFields(channelSnapInfoDetails{})
+// The fields we are interested in (not you, snap_yaml_raw)
+var detailFields = getStructFields((*snapDetails)(nil), "snap_yaml_raw")
```

Comment:
This is the motivation for the PR, right? Might be nice to put that into the PR description *why* this was changed from a struct to a pointer. Actually, is that really the reason? Or could we have the old way of looking at the struct just with exceptions?